### PR TITLE
rustc: add page

### PR DESCRIPTION
**Pull request by alx741**
The **Rust Compiler**
https://www.rust-lang.org/

### CODE REVIEW

File: pages/common/rustc.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+`rustc {{file.rs}}`
+
+- Compile with high optimization:
+
+`rustc -O {{file.rs}}`
+
+- Compile with debugging information:
+
+`rustc -g {{file.rs}}`
```
Comment (waldyrious) [inline pages/common/rustc.md]: Does the "g" stand for anything in particular?

File: pages/common/rustc.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- Build a test harness:
```
Comment (waldyrious) [inline pages/common/rustc.md]: Any chance we could use a more beginner-friendly description? Maybe a bit more descriptive would help.

**Comment by waldyrious** [comment]
Thanks for the contribution! Please check the inline comments.

File: pages/common/rustc.md (hunk 1 above)
Comment (alx741) [inline pages/common/rustc.md]: Actually *rustc* takes the `-g` parameter as an alias of `-C debuginfo=2`  so the former is kind of a useful layer for ease of use, and there is no expanded option like `--debug` or anything like that. Should I use `-C debuginfo=2`  instead?

Comment (waldyrious) [inline pages/common/rustc.md]: No, I don't think the extra complexity is worth it. I suppose one can hang on to the "g" in "debug" for a mnemonic, which should still be more memorable than `-C debuginfo=2`. Thanks for the explanation.

**Comment by alx741** [comment]
This one here has corrections to be reviewed

**Comment by waldyrious** [comment]
The description expansion looks good. Please check my comment about the "test harness" example.

File: pages/common/rustc.md (hunk 2 above)
Comment (alx741) [inline pages/common/rustc.md]: I was digging a bit more about it and the `--test` option is currently very poorly documented and is almost of none practical use for the user because **cargo** manages the testing. So If it is not practical i think it doesn't belong in the *tldr page*.

**Comment by alx741** [comment]
As i said in the inline comments, it shouldn't actually be in the *tldr page* so it's gone now.

**Comment by waldyrious** [comment]
👍, let's just wait for another maintainer to vet the PR, per our contribution policy. Thanks!
